Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,469 @@
+/-
+Copyright (c) 2024 Mario Carneiro and Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Emily Riehl
+-/
+
+import Mathlib.AlgebraicTopology.SimplicialSet.Nerve
+import Mathlib.CategoryTheory.Category.ReflQuiv
+import Mathlib.Combinatorics.Quiver.ReflQuiver
+
+
+/-!
+
+# The homotopy category of a simplicial set
+
+The homotopy category of a simplicial set is defined as a quotient of the free category on its
+underlying reflexive quiver (equivalently its one truncation). The quotient imposes an additional
+hom relation on this free category, asserting that `f ≫ g = h` whenever `f`, `g`, and `h` are
+respectively the 2nd, 0th, and 1st faces of a 2-simplex.
+
+In this file, we in fact define a pair of functors:
+
+(1) `SSet.hoFunctor' : SSet.{u} ⥤ Cat.{u, u}` implements the construction described above, while
+
+(2) `SSet.hoFunctor : SSet.{u} ⥤ Cat.{u, u} := SSet.truncation 2 ⋙ SSet.hoFunctor₂` is defined by
+first restricting from simplicial sets to 2-truncated simplicial sets (throwing away the data that
+is not used for the construction of the homotopy category) and then composing with an analogously
+defined `SSet.hoFunctor₂ : SSet.Truncated.{u} 2 ⥤ Cat.{u,u}` implemented relative to the syntax of
+the 2-truncated simplex category.
+
+It should be relatively straightforward to show that these constructions agree:
+
+def hoFunctor.ofTwoTruncation.iso (V : SSet) :
+    SSet.hoFunctor₂Obj ((truncation 2).obj V) ≅ SSet.hoCat V := sorry
+
+def hoFunctor.ofTwoTruncation.natIso :
+     truncation 2 ⋙ SSet.hoFunctor₂ ≅ SSet.hoFunctor' := sorry
+
+but we leave this for future work.
+
+The functor `SSet.hoFunctor` is shown to be left adjoint to the nerve by providing an analogous
+decomposition of the nerve functor and then composing a pair of adjunctions, which factor through
+the category of 2-truncated simplicial sets.
+-/
+
+namespace CategoryTheory
+open Category Limits Functor Opposite Simplicial
+universe v u
+
+section
+
+local notation3:1000 (priority := high) X " _[" n "]" =>
+    (X : CategoryTheory.SimplicialObject _).obj (Opposite.op (SimplexCategory.mk n))
+
+/-- A simplicial set `S` has an underlying refl quiver with `S _[0]` as its underlying type.-/
+def SSet.OneTruncation (S : SSet) := S _[0]
+
+/-- The source vertex of `f : S _[1]` for use in defining the underlying refl quiver.-/
+def SSet.OneTruncation.src {S : SSet} (f : S _[1]) : OneTruncation S :=
+  S.map (SimplexCategory.δ (n := 0) 1).op f
+
+/-- The target vertex of `f : S _[1]` for use in defining the underlying refl quiver.-/
+def SSet.OneTruncation.tgt {S : SSet} (f : S _[1]) : OneTruncation S :=
+  S.map (SimplexCategory.δ (n := 0) 0).op f
+
+/-- The hom-types of the refl quiver underlying a simplicial set `S` are subtypes of `S _[1]`.-/
+def SSet.OneTruncation.Hom {S : SSet} (X Y : OneTruncation S) :=
+  {p : S _[1] // src p = X ∧ tgt p = Y}
```

Comment:
It may be advisable to introduce a structure:
```lean
structure SSet.OneTruncation.Hom {S : SSet} (X Y : S _[0]) :=
  arrow : S _[1]
  src : src arrow = X := by aesop_cat
  tgt : tgt arrow = Y := by aesop_cat
```
Then, you may try to setup automation so that in most definitions, `src` and `tgt` are filled automatically.

Reply:
Unfortunately, this doesn't work, it breaks the proof by refl that the OneTruncation and OneTruncation₂ definitions are equivalent.